modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevocationWithSessionTerminationTestCase.java

```diff
@@ -0,0 +1,347 @@
+
+        OAuthConsumerAppDTO appDto = createApplication();
+        Assert.assertNotNull(appDto, "Application creation failed.");
+        consumerKey = appDto.getOauthConsumerKey();
+        Assert.assertNotNull(consumerKey, "OAuth clientId is invalid.");
+        consumerSecret = appDto.getOauthConsumerSecret();
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testRegisterApplication"}, description = "Test login using openId connect " +
+            "authorization code flow")
+    public void testOIDCLogin() throws Exception {
+
+        initiateAuthorizationRequest();
+        authenticateUser();
+        performConsentApproval();
+        generateAuthzCodeAccessToken();
+        introspectActiveAccessToken();
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testOIDCLogin"}, description = "Get User session using session termination REST API")
+    public void testGetUserSessions() {
+
+        Response response = getResponseOfGet(SESSION_API_ENDPOINT);
+        response.then()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .log().ifValidationFails()
+                .body("size()", notNullValue())
+                .body("userId", notNullValue())
+                .body("sessions", notNullValue())
+                .body("sessions.applications", notNullValue())
+                .body("sessions.applications.subject", notNullValue())
+                .body("sessions.applications.appName[0]", notNullValue())
+                .body("sessions.applications.appId", notNullValue())
+                .body("sessions.userAgent", notNullValue())
+                .body("sessions.ip", notNullValue())
+                .body("sessions.loginTime", notNullValue())
+                .body("sessions.lastAccessTime", notNullValue())
+                .body("sessions.id", notNullValue());
+        sessionIdList = response.jsonPath().getList("sessions.id");
+        Assert.assertEquals(sessionIdList.size(), 1);
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testGetUserSessions"}, description = "Terminate User session using session termination " +
+            "REST API")
+    public void testDeleteUserSessionById() {
+
+        String endpointURI = SESSION_API_ENDPOINT + "/" + sessionIdList.get(0);
+        // Delete the sessionId using session termination api.
+        getResponseOfDelete(endpointURI).then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_NO_CONTENT);
+        // Get all sessions.
+        getResponseOfGet(SESSION_API_ENDPOINT)
+                .then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("size()", is(0));
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testGetUserSessions"}, description = " The expected behaviour is, When the session is " +
```

Comment:
Shouldn't this be depending on `testDeleteUserSessionById` instead of `testGetUserSessions`